x/programs/examples/automated-market-maker/src/lib.rs

```diff
@@ -236,3 +236,56 @@ mod internal {
         );
     }
 }
+
+#[cfg(test)]
+mod tests {
```

Comment:
these tests are essentially the same as our current integration tests, could re-write as unit tests or update the integration tests to be more end to end style

Reply:
Should probably add a `#[ignore]` so it doesn't look like there are passing tests in CI